src/coreclr/nativeaot/Runtime/loongarch64/WriteBarriers.S

```diff
@@ -0,0 +1,354 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#include <unixasmmacros.inc>
+
+// Macro used to copy contents of newly updated GC heap locations to a shadow copy of the heap. This is used
+// during garbage collections to verify that object references where never written to the heap without using a
+// write barrier. Note that we are potentially racing to update the shadow heap while other threads are writing
+// new references to the real heap. Since this can not be solved perfectly without critical sections around the
+// entire update process, we instead update the shadow location and then re-check the real location (as two
+// ordered operations) and if there is a disparity we will re-write the shadow location with a special value
+// (INVALIDGCVALUE) which disables the check for that location. Since the shadow heap is only validated at GC
+// time and these write barrier operations are atomic wrt to GCs this is sufficient to guarantee that the
+// shadow heap contains only valid copies of real heap values or INVALIDGCVALUE.
+#ifdef WRITE_BARRIER_CHECK
+
+    .global     $g_GCShadow
+    .global     $g_GCShadowEnd
+
+        // On entry:
+        //  $destReg: location to be updated
+        //  $refReg: objectref to be stored
+        //
+        // On exit:
+        //  t3,t4: trashed
+        //  other registers are preserved
+        //
+        .macro UPDATE_GC_SHADOW destReg, refReg
+
+        // If g_GCShadow is 0, don't perform the check.
+        PREPARE_EXTERNAL_VAR_INDIRECT g_GCShadow, $t3
+        beq  $t3, $zero, 1f
```

Comment:
Can the branch instructions use a label?

Reply:
As these codes are within a macro, it is better using `1f` here.